.gitignore

```diff
@@ -8,3 +8,101 @@ Testing.java
 # remove emacs
 *~
 
+
+# Created by https://www.gitignore.io/api/intellij,intellij+iml
```

Comment:
What's the license of the output of gitignore.io? Also I don't think we really need all these ignores for intelliJ -- would not a one line ignore of `.idea/` be sufficient?